projects/packages/account-protection/src/class-password-manager.php

```diff
@@ -0,0 +1,198 @@
+
+/**
+ * Class Password_Manager
+ */
+class Password_Manager {
+	/**
+	 * Validaton service instance
+	 *
+	 * @var Validation_Service
+	 */
+	private $validation_service;
+
+	/**
+	 * Validation_Service constructor.
+	 *
+	 * @param ?Validation_Service $validation_service Password manager instance.
+	 */
+	public function __construct( ?Validation_Service $validation_service = null ) {
+		$this->validation_service = $validation_service ?? new Validation_Service();
+	}
+
+	/**
+	 * Verify the nonce for password update.
+	 *
+	 * @param string $key The nonce key.
+	 *
+	 * @return bool True if the nonce is valid, false otherwise.
+	 */
+	private function verify_password_update_nonce( $key ) {
+		if ( ! isset( $_POST['_wpnonce'] ) || ! wp_verify_nonce( sanitize_text_field( wp_unslash( $_POST['_wpnonce'] ) ), $key ) ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Verify the nonce for profile update.
+	 *
+	 * @param int $user_id The user ID.
+	 *
+	 * @return bool True if the nonce is valid, false otherwise.
+	 */
+	public function verify_profile_update_nonce( $user_id ) {
+		return $this->verify_password_update_nonce( 'update-user_' . $user_id );
+	}
+
+	/**
+	 * Validate the profile update.
+	 *
+	 * @param \WP_Error $errors The error object.
+	 * @param bool      $update Whether the user is being updated.
+	 * @param \stdClass $user A copy of the new user object.
+	 *
+	 * @return void
+	 */
+	public function validate_profile_update( \WP_Error $errors, bool $update, \stdClass $user ): void {
+		if ( ! $update ) {
+			// This is a new user (wp-admin/user-new.php)
+			if ( ! isset( $_POST['_new_user_nonce'] ) || ! wp_verify_nonce( sanitize_text_field( wp_unslash( $_POST['_new_user_nonce'] ) ), 'add-new-user' ) ) {
+				$errors->add( 'nonce_error', __( '<strong>Error:</strong> Nonce verification failed for new user creation.', 'jetpack-account-protection' ) );
+				return;
+			}
+			// This is an existing user update (wp-admin/profile.php or user-edit.php)
+		} elseif ( ! $this->verify_profile_update_nonce( $user->ID ) ) {
+			$errors->add( 'nonce_error', __( '<strong>Error:</strong> Nonce verification failed for profile update.', 'jetpack-account-protection' ) );
+			return;
+		}
+
+		if ( isset( $_POST['pass1'] ) && ! empty( $_POST['pass1'] ) ) {
+			$password = sanitize_text_field( wp_unslash( $_POST['pass1'] ) );
+
+			if ( $update ) {
+				$old_user_data = $this->get_old_user_data( $user->ID );
+				if ( $this->validation_service->is_current_password( $old_user_data, $password ) ) {
+					$errors->add( 'password_error', __( '<strong>Error:</strong> The password was used recently.', 'jetpack-account-protection' ) );
+					return;
+				}
+			}
+
+			$error = $this->validation_service->return_first_validation_error( $user, $password, 'profile' );
+			if ( ! empty( $error ) ) {
+				$errors->add( 'password_error', $error );
+				return;
+			}
+		}
+
+		// TODO: Run this even if JS validation passes?
```

Comment:
I assume we should always run the server side validation, these TODOs could likely be removed.

Reply:
[16d54a6](https://github.com/Automattic/jetpack/pull/41401/commits/16d54a6ff03ca1cbb08e798c97dcf7e670f0969f)

Reply:
@nateweller I just remembered something though, all our client side validation [will be by-passible](https://github.a8c.com/Automattic/waffle-makers-roadmap/issues/1#user-content-feature-password-strength) by selecting `Confirm use of weak password` so if we continue to run the server side validation, it'll still restrict them updating anyway. Any thoughts on how we should approach this?

Reply:
I think `$_POST['pw_weak']` is available to check against, which could be used to disable the appropriate checks on the server side 🤔

Reply:
Nice find! Added in [62d30c2](https://github.com/Automattic/jetpack/pull/41401/commits/62d30c290c700aa6b59a00c0f0218203a0648510)